subtests/docker_cli/systemd_in_container/test.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+#
+# Test systemd in containers
+#
+
+if [ $# -ne 1 ]; then
+    echo "Usage: $0 IMAGE-NAME"
+    exit 1
+fi
+image="$1"
+
+# Exit status. Default 0 (success), but set to 1 on any subtest failure
+rc=0
+
+# Main code. Run a command whose output is expected to be a single line,
+# and the first field is a security context (user:role:type:level); verify
+# that command generates exactly one line of output; extract the <type>
+# and <level> fields; and check against a list of one or more expected
+# values. Assume that there's no overlap between valid types & ranges.
+check_output() {
```

Comment:
>  Is the container name an issue? We can make it a parameter to the function.

If you look at my suggested clean rewrite, you'll see that container name is actually unnecessary. You can just run with `--detach` and get the CID. That way you don't have to worry about a subroutine improperly knowing the container name.

If you decide to go with a container name anyway, then yes, please, encapsulate it so that the subroutine finds out the name properly. And please remember to add a cleanup.